docs/infrastructure/fleet-on-kubernetes.md

```diff
@@ -0,0 +1,132 @@
+
+For the sake of this tutorial, we will use Helm, the Kubernetes Package Manager, to install MySQL and Redis. If you would like to use Helm and you've never used it before, you must run the following to initialize your cluster:
+
+```
+helm init
+```
+
+### MySQL
+
+The MySQL that we will use for this tutorial is not replicated and it is not Highly Available. If you're deploying Fleet on a Kubernetes managed by a cloud provider (GCP, Azure, AWS, etc), I suggest using their MySQL product if possible as running HA MySQL in Kubernetes can be difficult. To make this tutorial cloud provider agnostic however, we will use a non-replicated instance of MySQL.
+
+To install MySQL from Helm, run the following command. Note that there are some options that are specified. These options basically just enumerate that:
+
+- There should be a `kolide` database created
+- The default user's username should be `kolide`
+
+```
+helm install \
+  --name fleet-database \
+	--set mysqlUser=kolide,mysqlDatabase=kolide \
+  stable/mysql
+```
+
+This helm package will create a Kubernetes `Service` which exposes the MySQL server to the rest of the cluster on the following DNS address:
+
+```
+fleet-database-mysql:3306
+```
+
+We will use this address when we configure the Kubernetes deployment and database migration job, but if you're not using a Helm-installed MySQL in your deployment, you'll have to change this in your Kubernetes config files.
+
+#### Database Migrations
+
+The last step is to run the Fleet database migrations on your new MySQL server. To do this, run the following:
+
+```
+kubectl create -f ./examples/kubernetes/fleet-migrations.yml
+```
+
+In Kubernetes, you can only run a job once. If you'd like to run it again (i.e.: you'd like to run the migrations again using the same file), you must delete the job before re-creating it. To delete the job and re-run it, you can run the following commands:
+
+```
+kubectl delete -f ./examples/kubernetes/fleet-migrations.yml
+kubectl create -f ./examples/kubernetes/fleet-migrations.yml
+```
+
+### Redis
+
+```
+helm install \
+  --name fleet-cache \
+	--set persistence.enabled=false \
+  stable/redis
+```
+
+This helm package will create a Kubernetes `Service` which exposes the Redis server to the rest of the cluster on the following DNS address:
+
+```
+fleet-cache-redis:6379
+```
+
+We will use this address when we configure the Kubernetes deployment, but if you're not using a Helm-installed Redis in your deployment, you'll have to change this in your Kubernetes config files.
+
+## Setting Up and Installing Fleet
+
+### Create Server Secrets
+
+It should be noted that by default Kubernetes stores secret data in plaintext in etcd. Using an alternative secret storage mechanism is outside the scope of this tutorial, but let this serve as a reminder to secure the storage of your secrets.
+
+#### TLS Certificate & Key
+
+To generate a TLS key and a CSR (certificate signing request), you can run the following:
```

Comment:
Recommend `lego` with the DNS challenge here. https://github.com/xenolf/lego#cli-example Free and easy to potbain certs.

Reply:
Handy command I use:

```
GCE_PROJECT="kolide" GCE_DOMAIN="acme-co" \
    lego --email="victor@acme.co" \
    -x "http-01" \
    -x "tls-sni-01" \
    --domains="fleet.acme.co" \
    --dns="gcloud" --accept-tos run
```

The AWS one is very similar except for the env vars. There's also a handy k8s operator that manages certs for you. https://github.com/PalmStoneGames/kube-cert-manager

Worth mentioning.